src/masternode-sync.cpp

```diff
@@ -83,7 +83,7 @@ void CMasternodeSync::Reset()
 void CMasternodeSync::AddedMasternodeList(uint256 hash)
 {
     if(mnodeman.mapSeenMasternodeBroadcast.count(hash)) {
-        if(mapSeenSyncMNB[hash] < MASTERNODE_SYNC_THRESHOLD) {
+        if(mapSeenSyncMNB.count(hash) && mapSeenSyncMNB[hash] < MASTERNODE_SYNC_THRESHOLD) {
```

Comment:
This approach requires 2 map lookups to query a single value.

I would normally recommend instead:

<map iterator type> it = mapSeenSyncMNB.find(hash);
if((it != mapSeenSyncMNB.end()) && (it->second < MASTERNODE_SYNC_THRESHOLD))

In C++0x and later you can just use auto for the iterator type, otherwise I declare typedefs for most map iterator types, especially if they're used more than once.

That was my first impression but on taking a deeper look at this code I think neither your fix nor mine is correct.

There's no else clause on this conditional so if the key isn't found it doesn't get added.

The original code always adds the hash to the map if it doesn't exist, the problem is that it in that case it's doing a comparison with a value that may be uninitialized (since it's a plain C type, not a class).

In any case I doubt this is the cause of the crash.  I don't have a way to get line numbers from the mac crash report but if you've found a crash related to these maps my bet would be on the fact that they're being accessed without a mutex.  Unless you're certain that all accesses to these maps occur from a single thread they should be protected with a mutex.  And IMO, they should be made private and accessor methods added to the class so that all locking is handled within the class.